lib/src/kvs/ds.rs

```diff
@@ -1292,3 +1383,78 @@ impl Datastore {
 		Ok(())
 	}
 }
+
+async fn catchup_live_queries(
+	mut tx: Transaction,
+	local_live_queries: Arc<RwLock<BTreeMap<LqIndexKey, LqIndexValue>>>,
+	notification_channel_sender: Sender<Notification>,
+	lqs_to_update: Vec<(LqIndexKey, LqIndexValue)>,
+	vs: Versionstamp,
+) {
+	for (k, v) in lqs_to_update {
+		// Process notifications
+		let changes = {
+			let changes = cf::read(
+				&mut tx,
+				&k.ns,
+				&k.db,
+				Some(&k.tb),
+				ShowSince::Versionstamp(conv::versionstamp_to_u64(&v.vs)),
+				Some(1000),
+			)
+			.await;
+			tx.cancel().await.expect("Error cancelling transaction for lq catchup");
+			changes.expect("Error reading change feed")
+		};
+
+		for database_change in changes {
+			for table_changes in database_change.1 .0 {
+				for table_change in table_changes.1 {
+					// TODO enforce security
+					let not_type = match table_change {
+						TableMutation::Set(_thing, value) => {
+							// Atm all sets are UPDATE
+							Some((dbs::Action::Update, Some(value)))
+						}
+						TableMutation::Del(_thing) => Some((dbs::Action::Delete, None)),
+						TableMutation::Def(_stm) => None,
+					};
+					// We don't want to handle definition statements; Probably can be refactored better
+					if let Some((action, value)) = not_type {
+						let val = value.unwrap_or(Value::None);
+						notification_channel_sender
+							.send(Notification {
+								id: k.lq,
+								action,
+								result: val,
+							})
+							.await
+							.unwrap();
```

Comment:
Is a closed sender unexpected? Shouldn't we log this and return from the function instead?

Reply:
Logged